### PR TITLE
[cargo] use unstable cargo feature resolver.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Update docker/coverage to use the new cargo feature resolvers

### Have you read the [Contributing Guidelines on pull requests]

y

## Test Plan

ci -- all the ci's

Coverage with latest tooling:
https://app.circleci.com/pipelines/github/libra/libra/24553/workflows/4144e7db-26e7-417a-9128-25fac7bbc8ec/jobs/157384/steps

Docker build with latest tooling:
https://app.circleci.com/pipelines/github/libra/libra/24554/workflows/2f99c8ca-e0aa-4c48-a6a3-43dc16654d6c/jobs/157391

## Related PRs

None

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -65,11 +67,12 @@ commands:
       - run:
           name: Set cargo Environment
           command: |
+            # https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#features
+            export CARGOFLAGS=$(cat cargo-flags)
```
Comment (rexhoffman) [inline .circleci/config.yml:71]: I did the same thing with the cargo-flags as the cargo-toolchain, at least until we have an xbuild.

File: .circleci/config.yml [hunk 2]
```diff
@@ -624,6 +627,7 @@ jobs:
           command: |
             export RELEASE_BRANCH_PATTERN='^v[0-9]+-release$'
             export TEST_BRANCH_PATTERN="^testflow[0-9]+$"
+            export PRE_BRANCH_PATTERN="^premainnet-[0-9|.]+$"
```
Comment (rexhoffman) [inline .circleci/config.yml:630]: I also decided to pre-emptively support premainnet for docker image production in git (I'll need this Monday morning).   Easier to test out over the weekend.

File: .circleci/config.yml [hunk 3]
```diff
@@ -622,18 +625,20 @@ jobs:
...
             fi
             curl -o /tmp/pr https://api.github.com/repos/libra/libra/pulls/${pr_num}
-            export TARGET_BRANCH=$( cat /tmp/pr | jq ".base .ref" | sed 's/"//g' )
-            if [[ $TARGET_BRANCH =~ $RELEASE_BRANCH_PATTERN ]] || [[ $TARGET_BRANCH =~ $TEST_BRANCH_PATTERN ]]; then
-              export BRANCH=$TARGET_BRANCH
+            export BRANCH=$( cat /tmp/pr | jq ".base .ref" | sed 's/"//g' )
+
+            export IS_RELEASE=$( if [[ "$BRANCH" =~ "^v[0-9|.]+-release$" ]]; then echo true; else echo false; fi )
+            export IS_TEST=$( if [[ "$BRANCH" =~ "^testflow[0-9|.]+$" ]]; then echo true; else echo false; fi )
+            export IS_RRE_RELEASE=$( if [[ "$BRANCH" =~ "^premainnet-[0-9|.]+$" ]]; then echo true; else echo false; fi )
```
Comment (sausagee) [inline .circleci/config.yml:639]: RRE typo? Should it `IS_PRE_RELEASE`?

File: docker/build-common.sh [hunk 4]
```diff
@@ -7,7 +7,12 @@ set -e
 # It unifies RUSFLAGS, compilation flags (like --release) and set of binary crates to compile in common docker layer
 
 export RUSTFLAGS="-Ctarget-cpu=skylake -Ctarget-feature=+aes,+sse2,+sse4.1,+ssse3"
-cargo build --release -p libra-genesis-tool -p libra-node -p cli -p config-builder -p libra-key-manager -p safety-rules -p db-bootstrapper -p backup-cli -p cluster-test "$@"
+
```
Comment (sausagee) [inline docker/build-common.sh:10]: Some folks may get confused on the `RUST_NIGHTLY` and interpreting as we are building the binaries with the nightly Rust compiler toolchain. Can we put a note for clarification?

**Pull request closed without merging**